front/app/components/IdeaCards/shared/Filters/SortingBox/SortTypeButton.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
+
+import { Box, Text, Icon, colors } from '@citizenlab/cl2-component-library';
+import { useTheme } from 'styled-components';
+
+import { FrontOfficeSortOptions } from 'api/ideas/types';
+
+import { useIntl } from 'utils/cl-intl';
+
+import {
+  getButtonBackgroundColor,
+  getIconNameForSortingOption,
+  getLabelForSortingOption,
+} from './utils';
+
+type SortTypeButtonProps = {
+  sortType: FrontOfficeSortOptions;
+  handleSortOnChange: (sort: string) => void;
+  isSelected: boolean;
+};
+
+const SortTypeButton = ({
+  sortType,
+  handleSortOnChange,
+  isSelected,
+}: SortTypeButtonProps) => {
+  const theme = useTheme();
+  const { formatMessage } = useIntl();
+  const [isHover, setIsHover] = useState(false);
+
+  return (
```

Comment:
A couple of issues here:
- Performance-wise, by setting `isHover` we're re-rendering this button every time we enter or leave it. Hover state is covered well by CSS (and our `Button` component)
- The button is missing the pointer cursor, padding, etc.
- We should therefore use our `Button` from the component library, which will solve the above issues. It has props for background color on hover, built-in padding, cursor: pointer, etc.
- I prefer methods rather than inline functions for the handlers, especially in small components that can re-render often (inline functions get recreated on each re-render).